nautilus_open_any_terminal/open_any_terminal_extension.py

```diff
@@ -291,7 +291,11 @@ def _open_terminal(self, file_):
             if result.port:
                 value = "{0} -p {1}".format(value, result.port)
             if file_.is_directory():
-                value = '{0} cd "{1}" \\; $SHELL'.format(value, result.path)
+                path = unquote(result.path)
+                path = path.replace(" ", "\\ ")
+                path = path.replace("'", "\\'", -1)
+                path = path.replace('"', r'\\\\\\\"', -1)
```

Comment:
thanks for your PR.

to be honest I do not understand why this is needed... Could you please explain why we need this many escape characters?

additionally: please fix the CI.

Reply:
After a good night's rest I realized that since I'm escaping the spaces, I don't need quotes around the `cd` command anymore, so it doesn't add the complexity of all those escapes for `"`. Will create a new PR with cleaned and updated code.

Reply:
Because the command ends up being wrapped in double quotes `"` we need to escape that with two backslashes, so we need to write 4 of them

After which bash evaluates it to  (terminal, ip, and folder location varying of course)
```bash
terminator -e 'ssh -t 10.0.0.2 cd /home/IvoryJam/test\\\"ing  \; $SHELL'
``` 

Which ends up being ran as `cd /home/IvoryJam/test\"ing/` on the remote server

If we were to wrap the whole command in single quotes `'` it would work but we'd haven to do the same thing for single quotes (which we may want to escape the `$SHELL` portion because it's evaluating that variable on the local machine instead of  remotely and could break the workflow if the same shell isn't installed on the remote host)

Then the CI problem is for linting and it's just because I wrapped the string in single quotes fore readability but I can make that change

Reply:
$SHELL isn't evaulated on mine because I escaped it in my testing, normally it would be

```bash
terminator -e 'ssh -t 10.0.0.2 cd /home/IvoryJam/test\\\"ing  \; /bin/bash' 
```